vm/src/stdlib/os.rs

```diff
@@ -1191,6 +1191,55 @@ fn os_chmod(
     Ok(())
 }
 
+#[cfg(unix)]
+fn os_execv(
+    path: PyStringRef,
+    argv_list: PyIterable<PyStringRef>,
+    vm: &VirtualMachine,
+) -> PyResult<()> {
+    // TODO: argv_list to only accept tuples and lists
+    // message: "TypeError: execv() arg 2 must be a tuple or list"
+
+    if path.to_string().contains("\0") {
+        return Err(vm.new_value_error("embedded null byte".to_owned()));
+    }
+
+    let path = ffi::CString::new(path.as_str()).unwrap();
```

Comment:
? operator is used for Result enum in this function, so I'll probably use match instead of closure with map_err(). Thanks for the insight

Reply:
That's exactly what we want here. https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err

`O` in `map_err` takes `E` returns `F`. In this case, it can take `NulByteError` and returns `PyObjectRef` which is used for error type in RustPython.